### PR TITLE
[FIX] ScatterPlot Crashes on Data With Infinity Values

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Fixes # #2684 .

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/utils/scaling.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _compute_scaled_data(self):
         Y = data.Y if data.Y.ndim == 2 else np.atleast_2d(data.Y).T
         self.original_data = np.hstack((data.X, Y)).T
         self.scaled_data = no_jit = self.original_data.copy()
-        self.valid_data_array = ~np.isnan(no_jit)
+        self.valid_data_array = ~np.isnan(no_jit) & ~np.isinf(no_jit)
```
Comment (kernc) [inline Orange/widgets/utils/scaling.py]:
```py
self.valid_data_array = np.isfinite(no_jit)
```

Comment (jerneju) [reply Orange/widgets/utils/scaling.py]: Done.

File: Orange/widgets/tests/base.py [hunk 2]
```diff
@@ -813,3 +813,9 @@ def data_one_column_nans(cls):
                 "ynyn"
             )))
         return table
+
+    @classmethod
+    def data_one_column_infs(cls):
+        table = cls.data_one_column_nans()
+        table.X[:, 1] = np.inf
+        return table
```
Comment (kernc) [inline Orange/widgets/tests/base.py]: Could extend `data_one_column_nans()` into `data_one_column_vals(value=np.nan)` and call as `data_one_column_vals(np.inf)`.

Comment (jerneju) [reply Orange/widgets/tests/base.py]: Sure.

**Pull request merged**